### PR TITLE
Add questions bd

### LINKED ISSUE #23: Add endpoints for questions

### LINKED ISSUE #24: Add testing for questions files

### LINKED ISSUE #25: Add all relevant questions files

### PR DESCRIPTION
**Pull request by BlakeDonn**
#### What's this PR do?   

- Moves our questions dataset from the frontend to the backend
- Adds the migration, data, and seed files for the questions
- Adds questions GET endpoint to our express server 
- Adds testing that database functionality works as expected in the endpoint

#### Where should the reviewer start?    

- Start by rolling back migrations and updating to lates

#### How should this be manually tested?  

- Run `npm test` to check that the new testing block is passing
- Once migrated latest, re-seed and spin the server up
- Interact with the new endpoint via postman and check Postico respectively

#### Any background context you want to provide?  

- Was debating between determining the logic on the backend but decided against it

#### What are the relevant tickets? 

closes #23 
closes #24 
closes #25 

#### Screenshots (if appropriate) 

Screen Shot 2020-11-02 at 7 20 41 PM

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -92,12 +92,21 @@ app.get('/api/v1/podcasts/:genre', async (req, res) => {
...
     }
-  } catch(e) {
+  } catch (e) {
     res.status(500).json({e})
   }
 })
 
-module.exports = app;
\ No newline at end of file
+app.get('/api/v1/questions', async (req, res) => {
```
Comment (nicolegooden) [inline app.js]: Thanks for adding this endpoint!

File: db/data/questions.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
+const questionsData = [
```
Comment (nicolegooden) [inline db/data/questions.js]: Thanks for adding this in.  We should make sure we change `questionSet` to `questionsData` in our `Form` import, after we've made a request to retrieve this array.

**Review by nicolegooden — APPROVED** [review]
Thank you so much for getting this handled!  I think this is a good call to transfer the data from `questions.js` on the frontend side.  Also, the testing looks great, I appreciate the upkeep with that as additions were made to the database.

**Comment by BlakeDonn** [comment]
Check new commits:
* renamed most files to follow a consistent convention.
* ran linter on all files
* removed unnecessary variables
* changed activities to a string for questions data
* changed movie genres to an array

**Comment by BlakeDonn** [comment]
* renamed all `activities` keys to activity to match the frontend

**Comment by nicolegooden** [comment]
@BlakeDonn  Are we supposed to be putting complex data types in our tables? I thought Leta mentioned we shouldn't use arrays or objects, but could be wrong/misunderstood

File: app.js [hunk 3]
```diff
@@ -1,103 +1,118 @@
...
+app.get("/api/v1/questions", async (req, res) => {
+  try {
+    const questions = await database("questions").select();
+    res.status(200).json(questions);
+  } catch (e) {
+    res.status(500).json({e});
+  }
+});
+
+module.exports = app;
```
Comment (Atos20) [inline app.js:118]: Thanks for removing the unnecessary spacing, and also, should we make use of double-quotes standard from now?

File: db/data/board_games_data.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    const gamesInfo = await apiCalls.boardGame(games[counter])
+    if (gamesInfo) {
+      gamesToPost.push(gamesInfo.games[0])
+    }
+    counter++
+    setTimeout(() => {console.log(`${counter} / 20`)}, 100)
+  }
+  return gamesToPost
+}
+
```
Comment (Atos20) [inline db/data/board_games_data.js:17]: this feels like hacking the server, pretty neat!

File: db/data/movies_data.js [hunk 5]
```diff
@@ -1,6 +1,6 @@
 const apiCalls = require("../../apiCalls.js")
 
-const movieData = async (genre) => {
+const movie_data = async (genre) => {
```
Comment (Atos20) [inline db/data/movies_data.js:3]: I like this naming convention

File: db/data/questions_data.js [hunk 6]
```diff
@@ -0,0 +1,112 @@
+const questions_data = [
```
Comment (Atos20) [inline db/data/questions_data.js:1]: Nice addition!

**Review by Atos20 — COMMENTED** [review]
Thanks for the hard work!

**Pull request merged**